apps/daimo-mobile/src/view/screen/NoteScreen.tsx

```diff
@@ -0,0 +1,204 @@
+
+  useEffect(() => {
+    (async () => {
+      const note = await fetchNote(chain.clientL2, ephemeralOwner);
+      setNote(note);
+      if (note) setLoadState("loaded");
+      else setLoadState("error");
+    })();
+  }, [ephemeralOwner]);
+
+  return (
+    <ScrollView contentContainerStyle={styles.vertOuter} bounces={false}>
+      {loadState === "loading" && <Spinner />}
+      {loadState === "error" && <ErrorDisplay />}
+      {loadState === "loaded" && note && (
+        <NoteDisplay
+          ephemeralNote={note}
+          ephemeralPrivateKey={ephemeralPrivateKey}
+        />
+      )}
+    </ScrollView>
+  );
+}
+
+function Spinner() {
+  return (
+    <View style={ss.container.center}>
+      <ActivityIndicator size="large" />
+    </View>
+  );
+}
+
+function ErrorDisplay() {
+  return (
+    <>
+      {/** TODO: improve message and reporting */}
+      <TextError>Invalid note. May have been claimed already.</TextError>
+    </>
+  );
+}
+
+function NoteDisplay({
+  ephemeralNote,
+  ephemeralPrivateKey,
+}: {
+  ephemeralNote: EphemeralNote;
+  ephemeralPrivateKey: `0x${string}` | undefined;
+}) {
+  const [account] = useAccount();
+  assert(account != null);
+
+  const chain = useContext(ChainContext);
+  assert(chain != null);
+
+  const [ephemeralSignature, setEphemeralSignature] =
+    useState<`0x${string}`>("0x");
+
+  useEffect(() => {
+    (async () => {
+      if (!ephemeralPrivateKey) {
+        // Must be sender themselves reclaiming
+        setEphemeralSignature(dummySignature);
+        return;
+      }
+      const ephemeralAccount = privateKeyToAccount(ephemeralPrivateKey);
+      const ephemeralClient = createWalletClient({
+        account: ephemeralAccount,
+        chain: baseGoerli,
+        transport: http(),
+      });
+      const message = keccak256(account.address);
+      const signature = await ephemeralClient.signMessage({
+        message: { raw: message },
+      });
+      setEphemeralSignature(signature);
+    })();
+  }, [ephemeralPrivateKey, ephemeralNote]);
+
+  const [senderName, setSenderName] = useState("");
+
+  useEffect(() => {
+    (async () => {
+      const name = await rpcFunc.resolveAddr.query({
+        addr: ephemeralNote.from,
+      });
+      if (name) setSenderName(name);
+    })();
+  }, [ephemeralNote]);
+
+  const createTxHandle = (
+    ephemeralOwner: Address,
+    signature: `0x${string}`
+  ): TxHandle => {
+    return async (account: DaimoAccount) => {
+      console.log(
+        `[ACTION] claiming note ${ephemeralOwner} ${signature} ${account}`
+      );
+      return account.claimEphemeralNote(ephemeralOwner, signature);
+    };
+  };
+
+  const { status, message, exec } = useSendAsync(
+    account.enclaveKeyName,
+    createTxHandle(ephemeralNote.owner, ephemeralSignature)
+  );
+
+  // TODO: load estimated fees
+  // NOTE: These fees seem difficult to communicate to user, think about it.
+  const fees = 0.05;
```

Comment:
re: communicating to user, my current best guess is
- if fee<$0.10, subtle, gray indicator under the send button that shows either (total you're paying) or (net you're receiving) after fees
- fee >= $0.10, indicator is still small but now a ⚠️ yellow caution indicator
- fee > $1.00 or simulation unavailable, ❗️red warning, send button disabled or behind a confirm popup

Reply:
not in this pr ofc. planning to make a reusable send-user-op component that handles (button, simulation/fee display, spinner, success, error state)